packages/contracts/deploy/verify.ts

```diff
@@ -3,71 +3,152 @@ import {DeployFunction} from 'hardhat-deploy/types';
 import {TASK_ETHERSCAN_VERIFY} from 'hardhat-deploy';
 
 import {verifyContract} from '../utils/etherscan';
+import {getContractAddress} from './helpers';
 
 function delay(ms: number) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
 const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  console.log('\nVerifying contracts');
+
   const {deployments, ethers, run} = hre;
 
-  console.log('Verifying registry and factories contracts');
+  const msDelay = 2; // 3 minutes - Etherscan needs some time to process before trying to verify.
 
   console.log(
-    'Waiting for 3 minutes so Etherscan is aware of contracts before verifying'
+    `Waiting for ${
+      msDelay / 60000
+    } minutes, so Etherscan is aware of contracts before verifying`
   );
-  await delay(180000); // 3 minutes - Etherscan needs some time to process before trying to verify.
-  console.log('Starting to verify now');
 
-  await run(TASK_ETHERSCAN_VERIFY, {
-    license: 'GPL-3.0',
-    solcInput: true,
-  });
+  await delay(msDelay);
 
-  const RegistryContract = await ethers.getContractAt(
+  // Prepare contracts and addresses
+  const managingDAOAddress = await getContractAddress('DAO', hre);
```

Comment:
wouldn't it be possible to get all the addresses from all contracts from the deployments folder? this way we cannot forget to add a contract to this verify list